### ISSUE TITLE
Sync calendar & contacts opens app store instead of DAVx5

### ISSUE THREAD

**Issue by Whisprin** [nextcloud/android#9339]
### Steps to reproduce
0. Have Nextcloud and DAVx5 installed
1. Open Nextcloud settings
2. Click on "Sync calendar & contacts"
3. Android Systems show a "Open with" dialog with different app stores [1]

### Expected behaviour
DAVx5 opens with the Nextcloud Login flow as shown here: https://www.davx5.com/tested-with/nextcloud

### Actual behaviour
Android opens an app store to install DAVx5 although it's already installed [2]. Clicking on "open" in the app stores launches DAVx5 but _doesn't_ start the Login flow.

### Can you reproduce this problem on https://try.nextcloud.com?
Yes

### Environment data
Android version: 11

Device model: SM-F711B

Stock or customized system: Stock

Nextcloud app version: 3.18.0 (Google Play)
DAVx5 app version: 4.0-ose (F-Droid)

Nextcloud server version: 21.0.4

I opened the same issue in the [DAVx5 forum](https://forums.bitfire.at/topic/2631/nextcloud-sync-setup-opens-app-store-but-davx5-is-installed).

### Screenshots
[1] ![screenshot_20211121-202242_android-system](https://user-images.githubusercontent.com/3153136/142782182-3560d715-3e58-44f8-a62f-6de11bf10ef4.jpg)

[2] ![screenshot_20211121-202436_f-droid](https://user-images.githubusercontent.com/3153136/142782186-71cff58e-dd91-4b6f-8c5e-41acd6431569.jpg)

**Comment by mwen** [comment]
I have the same issue as @Whisprin. When sidloading the 1.17.1 apk from F-Droid on my smartphone (OnePlus Nord), I can see the login flow is properly working while the latest Play Store version (1.18) indeed only shows the DAVX5 app in the Play Store without any kind of login flow available.

**Comment by Whisprin** [comment]
@mwen Which of the Apps did you downgrade? Do you mean Nextcloud 3.17.1?

**Comment by AndyScherzinger** [comment]
@AlvaroBrey like opening files, I guess this one is also due to the sdk30 changes.

**Comment by mwen** [comment]
@Whisprin Yes, I meant the Nextcloud app for Android (version 3.17.1).

**Comment by Rathmox** [comment]
I have the same issue on Android 12, I tried installing from F-droid and Play Store, and it doesn't work

**Comment by AlvaroBrey** [comment]
@Rathmox will be fixed in the upcoming 3.18.1 patch release

**Comment by youngage** [comment]
Is there any ETA on the 3.18.1 release? Nextcloud App is effectively broken for new users for a whole month now and this is really frustrating :(

If there is anything blocking the release could you maybe push/prioritize this fix?

**Comment by shankari** [comment]
FYI, I just installed 3.18.1 (release date: Dec 22, 2021) on a new phone and did not run into this problem.
Clicking on "sync calendar & contacts" opened the app store when DAVx5 was *not* installed (as expected).
It did open DAVx5 with the nextcloud login flow once it was installed.
However, the nextcloud login flow did not support using an access token for the password. I had to log in with my nextcloud password and then "give access to this app"
